write-fonts/src/tables/cmap.rs

```diff
@@ -237,6 +236,187 @@ impl Cmap {
         ))
     }
 }
+// a helper for iterating over ranges for cmap 4
+struct Format4Ranges<'a> {
+    mappings: &'a [(char, GlyphId)],
+    seg_start: usize,
+    currently_contiguous: bool,
+}
+
+#[derive(Clone, Copy, Debug)]
+struct Format4Segment {
+    // indices are into the source mappings
+    start_ix: usize,
+    end_ix: usize,
+    id_delta: Option<i32>,
+}
+
+impl Format4Segment {
+    fn len(&self) -> usize {
+        self.end_ix - self.start_ix + 1
+    }
+
+    // cost in bytes of this segment
+    fn cost(&self) -> usize {
```

Comment:
add a spec ref?

Reply:
there's no specific part of the spec that describes a segment, but I'll try to make this clearer..